implementations/java-openapiprocessor/Dockerfile

```
@@ -1,6 +1,9 @@
-FROM gradle:8.3.0-jdk11
-WORKDIR /app
+FROM gradle:8.3.0-jdk17 AS builder
+WORKDIR /opt/app
 COPY . .
 RUN gradle installDist --no-daemon
 RUN chmod +x ./build/install/bowtie/bin/openapiprocessor
-CMD ["/app/build/install/bowtie/bin/openapiprocessor"]
+
+FROM bellsoft/liberica-openjdk-alpine:17
+COPY --from=builder /opt/app/build /opt/app/build
+CMD ["./opt/app/build/install/bowtie/bin/openapiprocessor"]
```

Presumably this is trying to be `/opt/app/etc/...` right? And same below?

---

hmm, I don't know what you mean - `openapiprocessor` is a sh script produces by the build, which basically runs `java -jar` underneath

---

Isn't this going to look in `/usr/app/opt/app/build/...`? Is that where the build outputs?

---

ahh, now I get it. Not really experienced linux user but I thought that to run any file you need to start with `./` - is it gonna work without?

---

No, that's only if you want a path relative to the current working directory.

If the working directory is `/`, then `./foo/bar` is `/foo/bar`.

If the working directory is `/bar/baz`, then `./foo/bar/` is `/bar/baz/foo/bar`, so this works basically "accidentally" because the working directory happens to be `/` -- but you don't want that here, you've explicitly copied the binary to the path `/opt/app/build/whatever`, so that's always where you want to run, so you should just have:

```suggestion
CMD ["/opt/app/build/install/bowtie/bin/openapiprocessor"]
```

essentially.

---

nice, thanks for clarification - changing it now